pysoa/server/action/introspection.py

```diff
@@ -0,0 +1,148 @@
+        "can be passed a single action name to return information limited to that single action. Otherwise, it will "
+        "return information for all of the service's actions."
+    )
+
+    request_schema = fields.Dictionary(
+        {
+            'action_name': fields.UnicodeString(
+                min_length=2,
+                description='Specify this to limit your introspection to a single action. It will be the only action '
+                            'present in the `actions` response attribute. If the requested action does not exist, an '
+                            'error will be returned.',
+            ),
+        },
+        optional_keys=('action_name', ),
+    )
+
+    response_schema = fields.Dictionary(
+        {
+            'documentation': fields.Nullable(fields.UnicodeString(
+                description='The documentation for the server, unless `action_name` is specified in the request body, '
+                            'in which case this is omitted.',
+            )),
+            'action_names': fields.List(
+                fields.UnicodeString(description='The name of an action.'),
+                description='An alphabetized list of every action name included in `actions`.',
+            ),
+            'actions': fields.SchemalessDictionary(
+                key_type=fields.UnicodeString(description='The name of the action.'),
+                value_type=fields.Dictionary(
+                    {
+                        'documentation': fields.Nullable(
+                            fields.UnicodeString(description='The documentation for the action'),
+                        ),
+                        'request_schema': fields.Nullable(fields.Anything(
+                            description='A description of the expected request schema, including any documentation '
+                                        'specified in the schema definition.',
+                        )),
+                        'response_schema': fields.Nullable(fields.Anything(
+                            description='A description of the guaranteed response schema, including any documentation '
+                                        'specified in the schema definition.',
+                        )),
+                    },
+                    description='A introspection of a single action',
+                ),
+                description='A dict mapping action names to action description dictionaries. This contains details '
+                            'about every action in the service unless `action_name` is specified in the request body, '
+                            'in which case it contains details only for that action.',
+            ),
+        },
+        optional_keys=('documentation', ),
+    )
+
+    def __init__(self, server):
+        super(IntrospectionAction, self).__init__(server.settings)
```

Comment:
`.__init__(server.settings)` feels wrong

Reply:
Care to elaborate? I don't see what the issue is. Can't read your mind 😄.

Reply:
sorry, I meant that `IntrospectionAction` inherits from `Action` who's signature is
```
    def __init__(self, settings=None):
        self.settings = settings
```
meaning we should make it clear that the first argument is not `settings` you'd expect when instantiating the action
```
def __init__(self, server, settings=None):
    super(IntrospectionAction, self).__init__(settings=server.settings)
```

Reply:
Okay, I understand. You're concerned that the conflicting interfaces are confusing. I'll resolve that with documentation.